common/enableEpochs/enableEpochsHandler.go

```diff
@@ -0,0 +1,118 @@
+package enableEpochs
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/core/atomic"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+var log = logger.GetOrCreate("enableEpochsHandler")
+
+type enableEpochsHandler struct {
+	*flagsHolder
+	enableEpochsConfig config.EnableEpochs
```

Comment:
[comment]
There is an interface for `MaxNumNodesChangeEnableConfigProvider` in `feat/liquid-staking`. When that PR is merged in `development`, I think we should move it here as well. Perhaps leave a `//todo` for it?

Reply:
as discussed, perhaps it is better to keep them separate or completely integrate it into `enableEpochsHandler` with a `GetMaxNodesChangeConfigProvider` new method..to be determined at that moment